### ISSUE TITLE
Page jumps to the top each time a block is selected or added

### ISSUE THREAD

**Issue by violetlynx** [WordPress/gutenberg#21622]
Hi!
After I've added enough blocks (all of them are images with links to another pages) that they can't be displayed in one window and there is a need to scroll down to view blocks at the bottom of the page, every time I choose a block to edit or add another one the page jumps to the top. 

I've tried deactivating basically all plugins but that haven't affected the issue. 
Installing Gutenberg plugin solved the problem, but with the default editor it happens how I've described above.

Here is an example of the content (I show just two blocks, but the page contain several dozens of them and it should have a lot more when I am done):

<div class="wp-block-image is-style-default">
<figure class="alignleft is-resized"><a href="https://violetlynxdyeworks.com/colorways/by-colorway-type/gradient-colorways/by-sets/set-a-dance-of-ice-and-fire/colorways-a-dance-of-ice-and-fire/"><img class="wp-image-3934" src="https://violetlynxdyeworks.com/wp-content/uploads/2018/08/DSCN4190-1-c-300x257.jpg" alt="A Dance of Ice and Fire" width="150" height="129" /></a>
<figcaption>A Dance of Ice and Fire</figcaption>
</figure>
</div>

<div class="wp-block-image is-style-default">
<figure class="alignleft size-large is-resized"><a href="https://violetlynxdyeworks.com/colorways/by-colorway-type/gradient-colorways/by-sets/four-seasons-set/blazing-autumn/"><img src="https://violetlynxdyeworks.com/wp-content/uploads/2018/09/DSCN2399-1-c-300x257.jpg" alt="Blazing Autumn" width="150" height="129" /></a>
<figcaption>Blazing Autumn</figcaption>
</figure>
</div>

**To reproduce**
1. Go to a page with a lot of blocks like shown above.
2. Scroll down to a block at the bottom of the page.
3. Click on it

Also if I choose a block and than choose HTML-editor other images just stay on top of the code so this mode is completely unusable. Here is a screenshot of how is looks:

![screenshot1](https://user-images.githubusercontent.com/33310256/79370134-5a8bb900-7f7c-11ea-8579-4c190eaadd6f.jpg)

**Editor version (please complete the following information):**
- WordPress version 5.4
- Default Block editor

**Desktop (please complete the following information):**
 - Windows 10
 - Browser - Firefox ver. 72.0.2 and Chrome ver. 80.0.3987.163

**Comment by jasmussen** [comment]
Hello, thank you for the report!

@youknowriad do you know if this ticket will be fixed by #21460?

**Comment by youknowriad** [comment]
I think so, @violetlynx Can you confirm that the last Gutenberg plugin update fixes this for you?

**Comment by violetlynx** [comment]
Hi!

Yes, after I've installed the Gutenberg plugin there is no jumps to the top. Although there is one other problem with switching between html and visual (it crashes) I use a roundabout way of doing it.

**Comment by jasmussen** [comment]
> Although there is one other problem with switching between html and visual (the page crashes) unless I use a roundabout way of doing it.

Can you clarify what you mean by "crashes"? Are you getting any error messages you can share here?

**Comment by violetlynx** [comment]
Here is what I get as a result:
![error](https://user-images.githubusercontent.com/33310256/79453914-8eb2b880-8014-11ea-9837-199453fc4a91.jpg)
 
The sequence that lead to this:

1. I create new page, enter title.
2. Create first block (insert foogallery)
3. Create second block from "formattin\classic" menu.
4. Switch this block into HTML-view and enter content.
5. Try to switch back to Visual and the error happens.

What I have to do is after step 4 - publish the page, than reload and only after that I can work with this "classic" block in the visual editor.

**Comment by jasmussen** [comment]
Thanks @violetlynx. Can you click the "Copy error" button, and paste the contents here? That will help debugging.

**Comment by violetlynx** [comment]
388/Zs.Slot@https://violetlynxdyeworks.com/wp-content/plugins/gutenberg/build/block-editor/index.js?ver=9461ac3d0bdde79d58b6fd0b9c1d1b31:49:271620
je@https://violetlynxdyeworks.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:78:476
ph@https://violetlynxdyeworks.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:215:173
lh@https://violetlynxdyeworks.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:126:411
O@https://violetlynxdyeworks.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:121:71
ze@https://violetlynxdyeworks.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:118:14
mg/<@https://violetlynxdyeworks.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:53:49
unstable_runWithPriority@https://violetlynxdyeworks.com/wp-includes/js/dist/vendor/react.min.js?ver=16.9.0:26:340
Ma@https://violetlynxdyeworks.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:52:280
mg@https://violetlynxdyeworks.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:52:498
V@https://violetlynxdyeworks.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:52:431
Be@https://violetlynxdyeworks.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:119:77
xi@https://violetlynxdyeworks.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:39:175

**Comment by violetlynx** [comment]
To clarify, this doesn't happen when I use build-in editor, only when I use Gutenberg plugin.

**Comment by jasmussen** [comment]
Thank for the added info. I was unable to glean what might have caused that particular crash, I even tried retracing your steps with the foogallery plugin, but was unable to reproduce. 

Are you saying that you're using the custom HTML block and it's switching between the tabs that causes the crash?

<img width="385" alt="Screenshot 2020-04-16 at 15 43 10" src="https://user-images.githubusercontent.com/1204802/79463345-04a92680-7ff9-11ea-9697-06522af11326.png">

If that's the case then you might have entered some HTML that broke things. You don't have to share it in case it's sensitive, but if you could say it was an embed or what you were trying to accomplish, that might help.

**Comment by violetlynx** [comment]
I am not using a block for custom HTML but a classic block into which I add content as html (because when I've tried to copy text unto it in visual mode it looses colors I've assigned):
![error2](https://user-images.githubusercontent.com/33310256/79464870-d80b0400-8024-11ea-9a4e-6bea5db9a4fc.jpg)

And here is the content I enter into this block:
![error3](https://user-images.githubusercontent.com/33310256/79464926-e9eca700-8024-11ea-8d2b-2097a9a5d1d4.jpg)
I've tried to enter my text as a code but browser keep changing it into visual.

**Comment by violetlynx** [comment]
Crash happens when I try to swich back from html into visual.

**Comment by jasmussen** [comment]
Thanks for your patience. 

When you say this:

>  am not using a block for custom HTML but a classic block into which I add content as html

What exactly do you mean? Do you mean going into this code editor?

<img width="420" alt="Screenshot 2020-04-16 at 16 12 44" src="https://user-images.githubusercontent.com/1204802/79466641-21dff400-7ffd-11ea-9ac0-4d3521a60293.png">

Or do you mean this one?

<img width="349" alt="Screenshot 2020-04-16 at 16 13 20" src="https://user-images.githubusercontent.com/1204802/79466700-37551e00-7ffd-11ea-95e3-2cf0498639a8.png">

In both cases, I can't get it to crash, using the classic block.

I wanted to note that you can colorize text using the normal Paragraph block:

![text color](https://user-images.githubusercontent.com/1204802/79466858-6bc8da00-7ffd-11ea-9440-0aab957b12c3.gif)

Let me know if that helps.

**Comment by violetlynx** [comment]
In my previous message the fist screen was to show you which particular block I've used (it is in "formatting" sub-menu and called "classic"). 
And the crash happens when I try to switch the view only for the particular block (so I use the option on top of the page from your second example).

As for colors, yeah, I know that, but I don't want to do color change for several lines in several hundreds of blocks, so I just copy a "template" where all colors and formatting is made as I want it and just change few words, because for some reason when I copy text from one page into another in visual mode it looses all colors.

**Comment by violetlynx** [comment]
One more addition. It looks like it doesn't matter at all what I enter into the block, because even if I entered just a simple string of letters when I try to change from HTML mode into visual mode the page crashes.

**Comment by jasmussen** [comment]
I'm very much unsure what might cause this crash for you, I simply can't reproduce. Is your WordPress version up to date? Are all your plugins up to date? You could try a different browser, you could try disabling plugins one by one to try and figure out what might be causing it. It could even be a browser extension.

Regarding this:

> As for colors, yeah, I know that, but I don't want to do color change for several lines in several hundreds of blocks, so I just copy a "template" where all colors and formatting is made as I want it and just change few words, because for some reason when I copy text from one page into another in visual mode it looses all colors.

I'll create a new ticket to note that copy pasting from one paragraph to another should not strip out colors. That's definitely a bug.

In the mean time, and in absense of us figuring out what crashes the editor, you can do this:

![reusable](https://user-images.githubusercontent.com/1204802/79540594-b2174b00-8088-11ea-846d-d41d60f7cedf.gif)

That is, you create your layout as you want it, colors and everything. Then you save those blocks as **reusable**. Now whenever you need those colors again, you can insert the reusable block you saved, convert it to a regular block, and edit it. 

Hope that helps.

**Comment by jasmussen** [comment]
I created #21670 for copy pasting text color.

**Comment by violetlynx** [comment]
Thank you for your help! I understand it is difficult to find a bug when you can't check my end of things. I'll try to use reusable blocks and see if that will solv my immediate problem. Although the first thing - the jumping to the top in build-in editor - is more important (basically that is the only thing that prevents me from using it as I had no other problems their except loosing colors when copy things, but that is, as I understand, a bug that affect editor in general).

**Comment by jasmussen** [comment]
Thank you. Because the initial ticket, jumping, was fixed by updating the plugin, I'm going to close this one as addressed. If the crash keeps happening in next versions of the plugin or WordPress, or you find other steps to reproduce, come back and comment or open a new ticket and we'll look again.
